workloads/dataset-controller/dataset-controller.py

```diff
@@ -0,0 +1,52 @@
+import os
+
+from kubernetes import config, client, watch
+from prometheus_client import Counter, start_http_server
```

Comment:
In the future, our answer would be https://github.com/open-telemetry/opentelemetry.io. May consider thinking about it now, to be completed later.

Do you have any direct prometheus integration?

Reply:
I'll do OpenTelemetry if there's a good motivation, but Prometheus has been sufficient for me for years. This is sort of a minimal stub rather than a serious array of metrics.

Reply:
Essentially, prometheus has been broken into two parts. Telemetry capture happens via OpenTelemetry, and metrics visualization happens via Prometheus.

This is similar in nature to how dockerd was split into docker and runc. That split ocurred to reduce the risk of venture investment corrupting the container ecosystem.

 OpenTelemetry is a parallel situation.